### PR TITLE
reagents reaction() now has more application method

### LINKED ISSUE #8959: Can't acid objects on the floor
Previously, if you splashed acid on the floor, you would melt shit on that tile.
Now, it just makes a PILE OF ACID, and everything is untouched.

If you manually target an item, you can still melt them one at a time, but it's pretty hilarious when you're trying to melt a pile of 40 bullet casings.

### LINKED ISSUE #11053: #10244 has become a serious balance issue
https://github.com/tgstation/-tg-station/pull/10244
@Ikarrus 

I don't know if this was really well thought out, but the ability to transfer all reagents from a container by throwing it has made Chemistry/Medbay as a whole a bit on the wildly overpowered side. 

A morphine bottle (Of which there are quite a few in Medbay at roundstart) will overdose you, among other things - In fact, most bottles can hold enough to overdose or kill. Throwing is somewhat hard to dodge and bypasses shields among other things. Phlog bottles have come into wide use simply because without calomel once the bottle makes contact, you are guaranteed to be husked by the fire.

You claimed it wasn't OP, but I think that was just the result of insufficient foresight. Nobody can really defend this considering it is a:
- Unblockable 
- Harder to dodge than a projectile
- A syringe gun+ that makes the guns totally obsolete

Bio protections will stop that, but they are so niche and Chemistry is so easy to use that I think it's barely a justification, and asking anyone who's had hostile contact with medical lately will tell you the same thing I've written here.

This is less a symptom of the chemicals and more just that they were never intended to have such easy application at a distance. Syringes are 15u as a matter of balance. Being able to apply things like a Krokodil overdose at range is far too good.

Reagent transfers on their own need a revert.

### LINKED ISSUE #11149: A bunch of blob chems are absolutely broken
Sorium and darkmatter don't throw on expanding
Skin melter and boiling oil don't ignite
Probably a bunch more chems don't work right

### LINKED ISSUE #11208: Expanding onto a blobbernaut as blob attacks instead of expanding
Blobbernauts aren't immune to blob_act.

### PR DESCRIPTION
**Pull request by phil235**
- reagent reaction() now use more methods. Instead of just INGEST or TOUCH, we now have:
  - TOUCH (for splashing). It no longer transfer reagent by default, it's now only used for touch based effect like being flammable when touched by fuel or you and your stuff being acided when touched by acid). Fixes #11053 
  - INGEST (for injection, ingestion). No changes there.
  - PATCH (for patch application). It does no transfer of its own (but actual patches code do make a transfer) but is used for touch effect that ignore clothes and protection.
  - VAPOR (for smoke, foam and spray application (and blob attack). It does reagent transfer but it takes into account clothing protection
- Fixes a typo in humanoid/get_permeability_prot
- Fixes acid splashing on objects not destroying it. Fixes acid splashing on turf not destroying objects on the turf. Fixes #8959
- Changed damp rag smothering code a bit.
- Fixes blob sorium and dark matter vortex not working correctly. Fixes #11149
- Fixes blob expansion attacking its own blob mobs (blobbernaut,etc). Fixes #11208
- The effects of blob reagents from blob attack and blob smoke can now be mitigated by wearing permeable clothes such as hardsuits, medical clothes, masks. (the lower the permeability_coefficient the better). Changed blob smoke reagent amount from 5 to 10 to compensate so blob smoke effect aren't too affected by this.
- changed the volume argument of reaction_mob() to reac_volume to avoid confusion with the reagent var "volume".

### CODE REVIEW

**Comment by GunHog** [comment]
> Blob Smoke now doesn't do anything (the blob reagents have no code for method == VAPOR).

Why would you allow that to be intended? Why not just add the proper code for the blob reagents so that they work properly? If you meant this as a nerf, the way you are doing it makes it seem as if you are introducing a bug.

**Comment by Ergovisavi** [comment]
Shouldn't foam be TOUCH and smoke be VAPOR? Otherwise they're the same shit.

**Comment by phil235** [comment]
@GunHog It is an intended nerf, but if people want to keep blob smoke effect all I have to do is change `if(method ==PATCH)` to`if(method == PATCH || method == VAPOR)`

@Ergovisavi This PR doesn't change anything for foam or smoke, it's just a name change. Sure the VAPOR name isn't exactly a great fit for foam but the foam effects are unchanged.

**Comment by Iamgoofball** [comment]
@Ergovisavi Also, foam needs to be able to transfer in reagents as well.

File: code/modules/detectivework/footprints_and_rag.dm [hunk 1]
```diff
@@ -22,14 +22,17 @@
...
 	if(!proximity)
 		return
-	if(ismob(A) && A.reagents && reagents.total_volume)
-		var/mob/M = A
-		if(user.a_intent == "harm")
-			reagents.reaction(M, TOUCH)
-			reagents.clear_reagents()
+	if(iscarbon(A) && A.reagents && reagents.total_volume)
+		var/mob/living/carbon/C = A
+		if(user.a_intent == "harm" && !C.is_mouth_covered())
```
Comment (optimumtact) [inline code/modules/detectivework/footprints_and_rag.dm]: are there defines for intent type?

Comment (phil235) [inline code/modules/detectivework/footprints_and_rag.dm]: no, I don't think there is.

Comment (optimumtact) [inline code/modules/detectivework/footprints_and_rag.dm]: rest in peace, carry on then :+1:

**Comment by freerealestate** [comment]
If you're going to nerf blob, why not change it so its attacks don't ignore protection instead? Its attacks ignoring protection doesn't really make sense.

**Comment by phil235** [comment]
@freerealestate It's a question of balance. Do we really want every crewmember wearing a hardsuit or biosuit to be nearly immune to blob attacks?

**Comment by freerealestate** [comment]
Nope, but I don't think it should their attacks should ignore protection
entirely.

**Comment by phil235** [comment]
don't merge this yet, still got something to add.

**Comment by phil235** [comment]
Alright, I added the fixes of #11210 to this PR. I've also readded the blob reagent effects to blob smoke.
Also added blob effect protection with permeable clothes, see the OP.

**Pull request merged**